solutions/advanced/cf-891C

```diff
@@ -0,0 +1,150 @@
+---
+id: cf-891C
+source: CF
+title: Envy
+author: Chongtian Ma
+---
+
+[Official Analysis](https://codeforces.com/blog/entry/55841)
+
+## Explanation
+
+This problem requires some MST intuition. In Kruskal's algorithm, we are to sort the edges by non-decreasing weight, and try uniting vertices if they aren't already in the same component. Here, we can try to check every query using this algorithm, by processing each edge of each query offline.
+
+As we consider edges from non-decreasing weight, we can keep track of a partially-built MST. An edge can only be in a MST if we processed all edges that have weights less than this edge, and we are able to unite the endpoints of the edge. This is because edges that have the same weight are arbitrary, so if we prioritize considering this edge then another MST with the same total weight is constructed.
+
+Thus, for each query, we need all edges to satisfy this condition. We can process all edges in non-decreasing weight, and for each query, try to unite the endpoint of all the edges that have this weight. If any edge does not satisfy, the whole query fails.
+
+## Explanation
+
+**Rough Time Complexity:** $\mathcal{O}(q \log n)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp 
+#include <bits/stdc++.h>
+using namespace std;
+
+const int MAXN = 5e5 + 5;
```

Comment:
does this work w/o +5?

Reply:
want me to do +1?

Reply:
no, i want you to do +0

Reply:
i cant, unless you want me to put --s literally everywhere

Reply:
i don't even see a variable named s in your code

Reply:
i meant multiple "--"

Reply:
then do +1 and comment where needed "index 0 won't be used"